plugin/mouse/mouse.py

```diff
@@ -269,6 +275,10 @@ def show_cursor_helper(show):
 @ctx.action_class("user")
 class UserActions:
     def noise_trigger_pop():
+        # Pop can be configured to wake Talon, in which case we don't want to process it as a click.
```

Comment:
I think that doing stuff like this with if statements is an anti pattern in Talon. Preferably you should add a new context that matches sleep mode and add the action there.

Reply:
Sorry but I don't know how I should do that. Could you point me to where this has been done so I may adopt from there?

Reply:
Something like this
```python
ctx_sleep = Context()
ctx_sleep.matches = r"""
mode: sleep
"""

@ctx_sleep.action_class("user")
class UserActions:
    def noise_trigger_pop():
        # Pop can be configured to wake Talon, in which case we don't want to process it as a click.
        if setting_mouse_enable_pop_wake.get():
            actions.speech.enable()
```

Reply:
replaced if with this snipped - unsure where to put it though - added it to the very end of the file